azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -454,3 +539,73 @@ func resourceAzureRMKubernetesClusterServicePrincipalProfileHash(v interface{})
 
 	return hashcode.String(buf.String())
 }
+
+func base64Decode(str string) (string, bool) {
+	data, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return "", true
+	}
+	return string(data), false
+}
+
+func getKubeConfig(config *string) *KubeConfig {
+	if config == nil {
+		return nil
+	}
+
+	configStr, error := base64Decode(*config)
+	if error == false && configStr != "" {
+		log.Println(config)
+		var kubeConfig KubeConfig
+		err := yaml.Unmarshal([]byte(configStr), &kubeConfig)
+		if err == nil && len(kubeConfig.Clusters) > 0 && len(kubeConfig.Users) > 0 {
+			return &kubeConfig
+		}
+	}
+
+	return nil
+}
+
+//TODO: Hide these
```

Comment:
lol, agree with this comment

Reply:
heh - I'd suggest these would also be good candidates to move out to `./azurerm/helpers/kubernetes`?